### PR TITLE
Adds gw and remove ndmz namespace

### PR DESCRIPTION
**Pull request by ashraffouda**
### Description

Describe the changes introduced by this PR and what does it affect

### Changes

List of changes this PR includes

### Related Issues

List of related issues

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstring

### CODE REVIEW

File: cmds/modules/netlightd/main.go [hunk 1]
```diff
@@ -107,18 +108,29 @@ func action(cli *cli.Context) error {
...
 	}
 
-	_, err = resource.Create("dmz", bridge, &net.IPNet{
-		IP:   net.ParseIP("100.127.0.2"),
-		Mask: net.CIDRMask(16, 32),
-	}, netlight.NDMZGwIP, nil, myceliumSeedFromIdentity(identity.PrivateKey(cli.Context)))
-
+	// create mycelium for host
+	hostMyCelium := "hmycelium"
+	myBr := "mdmz"
```
Comment (xmonader) [inline cmds/modules/netlightd/main.go:118]: what is the m?

File: cmds/modules/netlightd/main.go [hunk 2]
```diff
@@ -107,18 +108,29 @@ func action(cli *cli.Context) error {
...
+	if !bridge.Exists(myBr) {
+		if _, err := bridge.New(myBr); err != nil {
+			return fmt.Errorf("couldn't create bridge %s: %w", myBr, err)
```
Comment (xmonader) [inline cmds/modules/netlightd/main.go:121]: could not

File: pkg/netlight/resource/mycelium.go [hunk 3]
```diff
@@ -178,17 +182,31 @@ func setupMycelium(netNS ns.NetNS, mycelium string, seed []byte) error {
...
+	if netNS != nil {
+
+		name := filepath.Base(netNS.Path())
+		if err := os.WriteFile(filepath.Join(MyceliumSeedDir, name), seed, 0444); err != nil {
+			return fmt.Errorf("failed to create seed file '%s': %w", name, err)
+		}
 
-	return ensureMyceliumService(zinit.Default(), name, list)
+		return ensureMyceliumService(zinit.Default(), &name, list)
+	} else {
```
Comment (Eslam-Nawara) [inline pkg/netlight/resource/mycelium.go:193]: you already return here, no need for else

File: cmds/modules/netlightd/main.go (hunk 1 above)
Comment (ashraffouda) [reply cmds/modules/netlightd/main.go:118]: it is for mycelium, I kept the old naming but we can rename it to `hmy-br` (host mycelium bridge) for example  will be better

File: pkg/netlight/resource/mycelium.go (hunk 3 above)
Comment (ashraffouda) [reply pkg/netlight/resource/mycelium.go:193]: how?

File: pkg/netlight/resource/mycelium.go [hunk 4]
```diff
@@ -178,17 +182,31 @@ func setupMycelium(netNS ns.NetNS, mycelium string, seed []byte) error {
...
 // Start creates a mycelium zinit service and starts it
-func ensureMyceliumService(z *zinit.Client, namespace string, peers []string) error {
-	zinitService := fmt.Sprintf("mycelium-%s", namespace)
+func ensureMyceliumService(z *zinit.Client, namespace *string, peers []string) error {
+	var zinitService string
+	if namespace != nil {
+		zinitService = fmt.Sprintf("mycelium-%s", *namespace)
+	} else {
+		zinitService = "mycelium-host"
+	}
```
Comment (Eslam-Nawara) [inline pkg/netlight/resource/mycelium.go:209]:
```suggestion
	zinitService := "mycelium-host"
	if namespace != nil {
		zinitService = fmt.Sprintf("mycelium-%s", *namespace)
	}
```

File: pkg/netlight/resource/mycelium.go (hunk 3 above)
Comment (Eslam-Nawara) [reply pkg/netlight/resource/mycelium.go:193]: the code in the else clause can be in the function body itself and we can remove the else bs

Comment (ashraffouda) [reply pkg/netlight/resource/mycelium.go:193]: ahha ok, I thought u meant remove the else code all together

**Review by xmonader — APPROVED** [review]

**Pull request merged**